controllers/provider-vsphere/pkg/webhook/controlplane/ensurer.go

```diff
@@ -0,0 +1,181 @@
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewEnsurer creates a new controlplane ensurer.
+func NewEnsurer(logger logr.Logger) genericmutator.Ensurer {
+	return &ensurer{
+		logger: logger.WithName("vsphere-controlplane-ensurer"),
+	}
+}
+
+type ensurer struct {
+	genericmutator.NoopEnsurer
+	client client.Client
+	logger logr.Logger
+}
+
+// InjectClient injects the given client into the ensurer.
+func (e *ensurer) InjectClient(client client.Client) error {
+	e.client = client
+	return nil
+}
+
+// EnsureKubeAPIServerDeployment ensures that the kube-apiserver deployment conforms to the provider requirements.
+func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, ectx genericmutator.EnsurerContext, dep *appsv1.Deployment) error {
+	template := &dep.Spec.Template
+	ps := &template.Spec
+	if c := extensionswebhook.ContainerWithName(ps.Containers, "kube-apiserver"); c != nil {
+		ensureKubeAPIServerCommandLineArgs(c)
+		ensureVolumeMounts(c)
+	}
+	ensureVolumes(ps)
+	return e.ensureChecksumAnnotations(ctx, &dep.Spec.Template, dep.Namespace)
+}
+
+// EnsureKubeControllerManagerDeployment ensures that the kube-controller-manager deployment conforms to the provider requirements.
+func (e *ensurer) EnsureKubeControllerManagerDeployment(ctx context.Context, ectx genericmutator.EnsurerContext, dep *appsv1.Deployment) error {
+	template := &dep.Spec.Template
+	ps := &template.Spec
+	if c := extensionswebhook.ContainerWithName(ps.Containers, "kube-controller-manager"); c != nil {
+		ensureKubeControllerManagerCommandLineArgs(c)
+		ensureVolumeMounts(c)
+	}
+	ensureKubeControllerManagerAnnotations(template)
+	ensureVolumes(ps)
+	return e.ensureChecksumAnnotations(ctx, &dep.Spec.Template, dep.Namespace)
+}
+
+func ensureKubeAPIServerCommandLineArgs(c *corev1.Container) {
+	c.Command = extensionswebhook.EnsureStringWithPrefixContains(c.Command, "--enable-admission-plugins=",
+		"PersistentVolumeLabel", ",")
+	c.Command = extensionswebhook.EnsureNoStringWithPrefixContains(c.Command, "--disable-admission-plugins=",
+		"PersistentVolumeLabel", ",")
+}
+
+func ensureKubeControllerManagerCommandLineArgs(c *corev1.Container) {
+}
+
+func ensureKubeControllerManagerAnnotations(t *corev1.PodTemplateSpec) {
+	t.Labels = extensionswebhook.EnsureAnnotationOrLabel(t.Labels, v1alpha1constants.LabelNetworkPolicyToPublicNetworks, v1alpha1constants.LabelNetworkPolicyAllowed)
+	t.Labels = extensionswebhook.EnsureAnnotationOrLabel(t.Labels, v1alpha1constants.LabelNetworkPolicyToPrivateNetworks, v1alpha1constants.LabelNetworkPolicyAllowed)
+	t.Labels = extensionswebhook.EnsureAnnotationOrLabel(t.Labels, v1alpha1constants.LabelNetworkPolicyToBlockedCIDRs, v1alpha1constants.LabelNetworkPolicyAllowed)
+}
+
+var (
+	cloudProviderConfigKubeControllerManagerVolumeMount = corev1.VolumeMount{
```

Comment:
As far as I see all of this volume mount stuff can be removed as it's not needed for vSphere.